### DISCUSSION TITLE
TerraZarf?

### DISCUSSION THREAD

**Discussion by anoncam** [zarf-dev/zarf#142] [Ideas]
Howdy partner!

I wanted to see if there are any infrastructure capabilities that are either planned for zarf or intentionally left out of.

The use case I am thinking of is:

Airgap deployment in which both the network infrastructure (routers, vpc, ACLs, etc) get provisioned along with underlying compute for k8s...then on top of that cluster, finally, the packages...basically what are your thoughts on baking in terraform modules that get laid out before any kubernetes things?

**Comment by jeff-mccoy** [comment]
Sure we need to just put an example together I think (or you can do a PR).  What I see today is you can easily include the terraform binary and any modules you need with zarf.  On the other side, the only missing step is some sort of concept of a generic exec function.  Like say you bring a script over that runs terraform with the rest of those assets, having a command in zarf to say "and then finally execute this script with these args" might be nice.

**Comment by anoncam** [comment]
Sweet I'll read through the codebase and make some comments to get the dialogue sharpened.

**Comment by anoncam** [comment]
This could be interesting
